build/lib/cs2solutions/duckiebot.py

```diff
@@ -0,0 +1,225 @@
+    number_ctr = len(t)
+    plt.figure(figsize=[14, 6])
+    # Plot the resulting trajectory (and some road boundaries)
+
+    plt.subplot(1, 4, 2)
+    plt.plot(y_coord_ref, x_coord_ref)
+    plt.legend(['reference'])
+    legend_list = ['reference']
+    for i in range(number_ctr):
+        plt.plot(y_ctr[i], x_coord_ref, linewidth=1)
+        legend_list.append(f'controller_{i}')
+
+    plt.legend(legend_list, loc='upper left')
+    plt.plot(y_coord_ref - 0.9/np.cos(theta_ref), x_coord_ref, 'k-', linewidth=1)
+    plt.plot(y_coord_ref - 0.3/np.cos(theta_ref), x_coord_ref, 'k--', linewidth=1)
+    plt.plot(y_coord_ref + 0.3/np.cos(theta_ref), x_coord_ref, 'k-', linewidth=1)
+
+
+
+    plt.xlabel('y [m]')
+    plt.ylabel('x [m]');
+    plt.axis('Equal')
+
+    # Plot the lateral position
+    plt.subplot(2, 2, 2)
+    plt.plot(t[0], y_coord_ref)
+    legend_list = ['reference']
+    for i in range(number_ctr):
+        plt.plot(t[i], y_ctr[i], linewidth=1)
+        legend_list.append(f'controller_{i}')
+    plt.legend(legend_list)
+    plt.ylabel('Lateral position $y$ [m]')
+
+    # Plot the control input
+    plt.subplot(2, 2, 4)
+    plt.plot(t[0], w_curvy)
+    for i in range(number_ctr):
+        plt.plot(t[i], w_ctr[i], linewidth=1)
+        legend_list.append(f'controller_{i}')
+    plt.legend(legend_list)
+    plt.ylabel('$\\omega$ [rad/s]')
+    plt.xlabel('Time t [sec]')
+    plt.tight_layout()
+
+# Utility function to plot the step response
+def plot_step_response(t: np.array, y: np.array, u: np.array):
+    axes_out = plt.subplot(2, 1, 1)
+    plt.sca(axes_out)
+    plt.plot(t, y)
+    plt.ylabel('Lateral position $y$ in [m]')
+    plt.plot([0, 20], [0.95, 0.95], 'k--')
+    plt.plot([0, 20], [1.05, 1.05], 'k--')
+
+    axes_input = plt.subplot(2, 1, 2)
+    plt.sca(axes_input)
+    plt.xlabel('Time t [sec]')
+    plt.ylabel('$\omega$ [rad/s]')
+    plt.plot(t, u[0])
+
+def plot_sim_results(y_ref: np.array, y: np.array,
+                     theta_ref: np.array, theta: np.array,
+                     t: np.array, w_ref: np.array,
+                     w: np.array) -> None:
+    # Configure matplotlib plots to be a bit bigger and optimize layout
+    plt.figure(figsize=[15, 4.5])
+
+    # Plot the actual and estimated states
+    ax = plt.subplot(1, 2, 1)
+    plt.plot(t, y)
+    plt.plot(t, y_ref, 'k-.')
+    ax.set(xlim=[0, 30])
+    plt.legend(['state feedback', 'y', 'reference'])
+    plt.xlabel('Time t [sec]')
+    plt.ylabel('Lateral position $y$ [m]')
+
+    ax = plt.subplot(2, 2, 2)
+    plt.plot(t, theta)
+    # plt.plot(t, x[3])
+    plt.plot(t, theta_ref, 'k-.')
+    ax.set(xlim=[0, 15])
+    plt.ylabel('Angle $\\theta$ [rad]')
+
+    ax = plt.subplot(2, 2, 4)
+    plt.plot(t, w)
+    # plt.plot(t, u_ofb[0])
+    plt.plot(t, w_ref, 'k-.')
+    ax.set(xlim=[0, 15])
+    plt.xlabel('Time t [sec]')
+    plt.ylabel('$\\omega$ [rad/s]')
+
+    
+
+# Default Duckiebot velocity in m/s
```

Comment:
maybe we can add all these global variables at the beginning of the file after the import section.